cloud-functions/index.js

```diff
@@ -50,6 +54,88 @@ app.get('/posts', (req, res, next) => {
     });
 });
 
+app.post('/create-user', jsonParser, (req, res, next) => {
+    const user = req.body;
+    const usersRef = db.collection('users');
+
+    const promises = [
+        usersRef.where('studentId', '==', user.studentId).get(),
+        usersRef.where('email', '==', user.email).get(),
+        usersRef.where('phone', '==', user.phone).get()
+    ];
+
+    Promise.all(promises)
+    .then((snapshots) => {
+        // given student id already exists
+        if (!snapshots[0].empty) {
+            responseStatus = 400;
+            responseObj = {
+                Body: null,
+                Message: 'A user with the given studentId already exists.',
+                Code: responseStatus
+            };
+            res.status(responseStatus).json(responseObj);
+            return;
+        }
+
+        // given email already exists
+        if (!snapshots[1].empty) {
+            responseStatus = 400;
+            responseObj = {
+                Body: null,
+                Message: 'A user with the given e-mail address already exists.',
+                Code: responseStatus
+            };
+            res.status(responseStatus).json(responseObj);
+            return;
+        }
+
+        // given phone number already exists
+        if (!snapshots[2].empty) {
+            responseStatus = 400;
+            responseObj = {
+                Body: null,
+                Message: 'A user with the given phone number already exists.',
+                Code: responseStatus
+            };
+            res.status(responseStatus).json(responseObj);
+            return;
+        }
+
+        usersRef.add({
+            name: user.name,
+            lastname: user.lastname,
+            phone: user.phone,
+            email: user.email,
+            studentId: user.studentId,
+            year: user.year,
+            department: user.department,
+            profilePhoto: user.profilePhoto
+        }).then(ref => {
+            responseStatus = 200;
+            responseMessage = `User added with ID: ${ref.id}`;
+            responseObj = {
+                Body: user,
```

Comment:
bu dondugumuz user bize gelen user. userRef bize bi girdi id si falan vermiyo mu ekstradan?

Reply:
bize ekledigi element'in id'sini return ediyor aslinda. oradan tekrar db sorgusu yapip, gelen user'daki fieldlari mi return edeyim?

Reply:
Sadece id mi geliyor? Tum modeli gondereymis iyiymis. normalde bu id ile islem yapmamiz lazim aslinda ama firebase auth un bildigi id de email olacak. Cok da onemli degil gibi bu durumda

Reply:
burada bize gelen user'i gondermemek lazim. user'i oldugu gibi geri donmek yerine user id donebiliriz. tercih meselesi gibi geliyor bana. bence id donelim

Reply:
```Body: {"userID": "1y2udsahhe12jhdsa"}``` bu document id bu arada

Reply:
bu document id'yi dondurmeyelim mi? yerine user document'lerinin icine id diye bir field gomup, id uretip onu dondurebiliriz. nasil bir yapi tavsiye edersin @gurolcay ?

Reply:
ya burda mantik aslinda db den indexli olan id sinin donmesini saglamak. ama burdaki db bildigimiz klasik db olmadigi icin cok da sorun yaratmiyo. Authenticationda da bu db kullanilmadigi icin sikinti yok. firestore gerekli indexlemeleri yapiyo demekki